### ISSUE TITLE
Function get_folded_lines() and is_line_folded() does not return the value if the line is folded

### ISSUE THREAD

**Issue by shinspiegel** [godotengine/godot#68949]
### Godot version

Godot 4 - beta 4

### System information

Macbook Pro (M1)

### Issue description

I'm updating a plugin that affects the code editor for the Godot, but the `get_folded_lines()` return the value only for the first line folded, but does not return the values for the other lines folded. And the function `is_line_folded()` does not return if the line if folded or not.

In the following example, lines 6~8 and 11~13 should be folded.

<img width="747" alt="Screenshot 2022-11-21 at 12 39 35" src="https://user-images.githubusercontent.com/30536099/203042896-af60667a-45ca-423a-ba31-e3213f06e36e.png">

But lines 5 and 10 are marked as folded. These are the initial lines, that hold the folded lines, but they are not hidden, just folded.

<img width="216" alt="Screenshot 2022-11-21 at 12 39 48" src="https://user-images.githubusercontent.com/30536099/203042915-dc361ce9-4ec6-41cf-ba48-cf2b125e7f7d.png">

I also tried to use the `is_line_hidden()` from the `TextEdit` class but looks like `CodeEdit` does not inherit this class anymore. Or does not have this method in this class.

### Steps to reproduce

1. Open the minimal project
2. Open the `testing.gd` script
3. Deactive and active the testing plugin
4. See the result in the output

### Minimal reproduction project

[folded_lines_minimal_project.zip](https://github.com/godotengine/godot/files/10056189/folded_lines_minimal_project.zip)

**Comment by akien-mga** [comment]
I think the result might be intentional - this API conveys the meaning of which line starts a folded scope. If all the lines in that scope would return `true` for `is_line_folded()`, you wouldn't know in this example if it's two folded blocks, or if line 9, 10, 11 and 12 are part of the block starting on line 5:

![image](https://user-images.githubusercontent.com/4701338/203059606-811735a9-1f34-4b2c-9957-f4a8c4f71238.png)

CC @Paulb23

**Comment by shinspiegel** [comment]
In this case, if I need to find the hidden lines I should use the `is_line_hidden()` method, but this is not available on the `CodeEdit` class.

**Comment by MewPurPur** [comment]
I think it makes sense to expose `is_line_hidden()` and related methods.

**Comment by shinspiegel** [comment]
I don't think this method exists, I'll check the source on the 3.5 branch and recreate it on the 4.0. 
But I need some guidance on the process, I'm no CPP developer, and that would be the first time I make any CPP code.
